foundation/gateway/pkg/adapter/kubernetes/labels/labels.go

```diff
@@ -0,0 +1,31 @@
+package labels
+
+import (
+	"crypto/sha3"
+	"fmt"
+)
+
+// ComputeKeyedLabelKey computes the keyed label key for a given original key.
+func ComputeKeyedLabelKey(key string) string {
+	return fmt.Sprintf("%s%x", KeyedLabelsPrefix, sha3.Sum224([]byte(key)))
+}
+
+func ComputeOriginalFromKeyedLabels(keyedLabels map[string]string, keys []string) map[string]string {
+	original := make(map[string]string, len(keys))
+	for _, key := range keys {
+		computedKey := ComputeKeyedLabelKey(key)
+		if value, exists := keyedLabels[computedKey]; exists {
+			original[key] = value
+		}
+	}
+	return original
+}
+
+func ComputeKeyedFromOriginalLabels(originalLabels map[string]string) map[string]string {
```

Comment:
```suggestion
func KeyedFromOriginal(originalLabels map[string]string) map[string]string {
```
Since it's already in the labels package?